### PR TITLE
Add fields to the Policy CRD for policy ordering

### PR DESCRIPTION
**Pull request by JustinKuli**
Refs:
 - https://github.com/open-cluster-management-io/enhancements/pull/74
 - https://github.com/stolostron/backlog/issues/26187

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

### CODE REVIEW

File: api/v1/policy_types.go [hunk 1]
```diff
@@ -24,25 +24,46 @@ const (
 // PolicyTemplate template for custom security policy
 type PolicyTemplate struct {
 	// +kubebuilder:pruning:PreserveUnknownFields
-	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+	ObjectDefinition  runtime.RawExtension `json:"objectDefinition"`
+	ExtraDependencies []PolicyDependency   `json:"extraDependencies,omitempty"`
+	IgnorePending     bool                 `json:"ignorePending,omitempty"`
```
Comment (mprahl) [inline api/v1/policy_types.go:29]: Could you document this field so it shows up in the CRD?

Comment (JustinKuli) [reply api/v1/policy_types.go]: Thanks for catching that! I also added some more comments for other fields, somewhat based on the doc at https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/governance/governance#policy-yaml-structure

File: api/v1/policy_types.go [hunk 2]
```diff
@@ -24,25 +24,58 @@ const (
 // PolicyTemplate template for custom security policy
 type PolicyTemplate struct {
 	// +kubebuilder:pruning:PreserveUnknownFields
+	// A Kubernetes object defining the policy to apply to a managed cluster
 	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+
+	// PolicyDependencies that only apply to this template
```
Comment (mprahl) [inline api/v1/policy_types.go:30]:
```suggestion
	// Additional PolicyDependencies that only apply to this template
```

File: api/v1/policy_types.go [hunk 3]
```diff
@@ -24,25 +24,58 @@ const (
...
 
-	// NonCompliant is an ComplianceState
+	// NonCompliant is a ComplianceState
 	NonCompliant ComplianceState = "NonCompliant"
+
+	// Pending is a ComplianceState
+	Pending ComplianceState = "Pending"
 )
 
+// Each PolicyDepenency defines an object reference which must be in a certain state
```
Comment (mprahl) [inline api/v1/policy_types.go:51]:
```suggestion
// Each PolicyDepenency defines an object reference which must be in a certain compliance state
```

**Comment by mprahl** [comment]
This looks great! Just a few nitpicks.

**Review by mprahl — APPROVED** [review]

**Comment by dhaiducek** [comment]
@JustinKuli Please also update the CRD in the addon controller, as the sync is currently failing:
https://github.com/open-cluster-management-io/governance-policy-addon-controller/actions/runs/3261608536/jobs/5356888725

**Pull request merged**